diff --git .config/zsh/aliases/shell.aliases.sh .config/zsh/aliases/shell.aliases.sh
index 77a3726..1d6f204 100644
--- .config/zsh/aliases/shell.aliases.sh
+++ .config/zsh/aliases/shell.aliases.sh
@@ -1,7 +1,7 @@
 #!/bin/bash
 
 # dotfiles aliases
-alias .f='cd ~/.config/zsh && cd -P ../..'
+alias .f='p && cd dotfiles'
 alias .fu='.f && git submodule update --init --recursive'
 alias .fe='.f && vim .'
 alias .fr='source ~/.zshrc; echo ".zshrc reloaded"'
diff --git .config/zsh/init/007_zoxide.init.sh .config/zsh/init/007_zoxide.init.sh
deleted file mode 100644
index 4baf6b1..0000000
--- .config/zsh/init/007_zoxide.init.sh
+++ /dev/null
@@ -1,5 +0,0 @@
-#!/bin/bash
-
-if type fzf >/dev/null 2>&1; then
-  eval "$(zoxide init --cmd cd zsh)"
-fi
diff --git .config/zsh/zshrc.zsh .config/zsh/zshrc.zsh
index e82c788..3e9b7ff 100644
--- .config/zsh/zshrc.zsh
+++ .config/zsh/zshrc.zsh
@@ -1,63 +1,64 @@
 #!/bin/bash
 
-# Get zgen
-source ~/.zgen/zgen.zsh
-
 export ZSH_CONFIG_HOME="$HOME/.config/zsh"
 export GPG_TTY=$TTY # https://unix.stackexchange.com/608921
 
-# Generate zgen init.sh if it doesn't exist
-if ! zgen saved; then
-  zgen oh-my-zsh
-  zgen oh-my-zsh plugins/genpass
-  zgen oh-my-zsh plugins/asdf
-  zgen oh-my-zsh plugins/tmux
+# Init zinit
+# 1. Set the directory we want to store zinit and plugins
+ZINIT_HOME="${XDG_DATA_HOME:-${HOME}/.local/share}/zinit/zinit.git"
+[ ! -d $ZINIT_HOME ] && mkdir -p "$(dirname $ZINIT_HOME)"
+# 2. Download Zinit, if it's not there yet
+[ ! -d $ZINIT_HOME/.git ] && git clone https://github.com/zdharma-continuum/zinit.git "$ZINIT_HOME"
+# 3. Source/Load zinit
+source "${ZINIT_HOME}/zinit.zsh"
 
-  zgen load agkozak/zsh-z
-  zgen load Aloxaf/fzf-tab
-  zgen load zsh-users/zsh-syntax-highlighting
-  zgen load zsh-users/zsh-history-substring-search
-  zgen load zsh-users/zsh-autosuggestions
-  zgen load zsh-users/zsh-completions
+# Add in zsh plugins
+zinit light zsh-users/zsh-syntax-highlighting
+zinit light zsh-users/zsh-completions
+zinit light zsh-users/zsh-autosuggestions
+zinit light zsh-users/zsh-history-substring-search
+# zinit light agkozak/zsh-z
 
-  # Generate init.sh
-  zgen save
-fi
+# Add in snippets
+zinit snippet OMZP::git
+zinit snippet OMZP::sudo
+zinit snippet OMZP::asdf
+zinit snippet OMZP::tmux
+zinit snippet OMZP::command-not-found
 
 # load completions
 autoload -U compinit && compinit
+zinit cdreplay -q
 
 # Completion styling
-zstyle ':completion:*' matcher-list 'm:{a-z}={A-Za-z}'
+zstyle ':completion:*' menu select
+zstyle ':completion:*' use-cache on
+zstyle ':completion:*' cache-path "$XDG_CACHE_HOME/zsh/.zcompcache"
+zstyle ':completion:*' file-list all
 zstyle ':completion:*' list-colors "${(s.:.)LS_COLORS}"
-zstyle ':completion:*' menu no
-zstyle ':fzf-tab:complete:cd:*' fzf-preview 'ls --color $realpath'
-zstyle ':fzf-tab:complete:__zoxide_z:*' fzf-preview 'ls --color $realpath'
+zstyle ':completion:*' matcher-list 'm:{a-z}={A-Za-z}'
 
 # History Options
+# Increase history size
+HISTSIZE=100000
+SAVEHIST=100000
+HISTFILE=~/.zsh_history
+export HISTIGNORE="&:ls:cd:pwd:exit:date:* --help"
+
 setopt append_history
 setopt extended_history
+setopt share_history # Share history across all your terminal windows
 setopt hist_expire_dups_first
 setopt hist_ignore_all_dups
 setopt hist_ignore_dups
 setopt hist_ignore_space
 setopt hist_reduce_blanks
 setopt hist_save_no_dups
+setopt hist_find_no_dups
 setopt hist_verify
-
-# Share history across all your terminal windows
-setopt share_history
-
-# set some more options
 setopt pushd_ignore_dups
 setopt pushd_silent
 
-# Increase history size
-HISTSIZE=100000
-SAVEHIST=100000
-HISTFILE=~/.zsh_history
-export HISTIGNORE="&:ls:cd:pwd:exit:date:* --help"
-
 # Return time on long running processes
 REPORTTIME=2
 TIMEFMT="%U user %S system %P cpu %*Es total"
